modules/python/init.zsh

```diff
@@ -92,18 +92,26 @@ if (( $+VIRTUALENVWRAPPER_VIRTUALENV || $+commands[virtualenv] )) && \
   # Disable the virtualenv prompt.
   VIRTUAL_ENV_DISABLE_PROMPT=1
 
-  # Enable 'virtualenv' with 'pyenv'.
-  if (( $+commands[pyenv] )) && \
-     pyenv commands | command grep -q virtualenv-init
-  then
+  # Create a sorted array of available virtualenv related 'pyenv' commands to
+  # look for plugins of interest. Scanning shell '$path' isn't enough as they
+  # can exist in 'pyenv' synthesized paths (e.g., '~/.pyenv/plugins') instead.
+  local -a virtenv_plugins
+  if (( $+commands[pyenv] )); then
+    virtenv_plugins=(${(@oM)${(f)"$(pyenv commands --no-sh 2>/dev/null)"}:#virtualenv*})
+  fi
+
+  if (( $+commands[pyenv] && \
+        $virtenv_plugins[(i)virtualenv-init] <= $#virtenv_plugins )); then
```

Comment:
@indrajitr if `virtenv_plugins` is "empty", this expression (`$virtenv_plugins[(i)virtualenv-init] <= $#virtenv_plugins`) returns False?

I'm asking it because if there is no `pyenv` installed, then `virtenv_plugins` is at its initial state (`local -a virtenv_plugins`). Then, may be we do not need to re-check for `$+commands[pyenv]` here right?